src/Aside/Aside.js

```diff
@@ -1,34 +1,53 @@
 import React, { Component } from 'react';
 import Controls from '../Controls/Controls.js';
 import AsideBody from '../AsideBody/AsideBody.js';
+import {cards} from '../mtg.js'
 import './Aside.css';
 
 class Aside extends Component {
   constructor() {
     super();
     this.state = {
-      cardNames: []
+      cardNames: [],
+      hasDuplicates: false,
+      isInvalidCardName: false
     };
   }
 
   addCardToList = (cardName) => {
-    const newState = {
-      cardNames: this.state.cardNames.concat([cardName])
-    };
-    this.setState(newState);
+    const matchingCard = cards.find(card => {
+      return card.cardName.toLowerCase() === cardName.toLowerCase();
+    });
+    if (!this.state.cardNames.includes(matchingCard.cardName)) {
+      const newState = {
+        cardNames: this.state.cardNames.concat([matchingCard.cardName]),
+        hasDuplicates: false,
+        isInvalidCardName: false
+      };
+      this.setState(newState);
+    } else {
+      this.setState({ hasDuplicates: true });
```

Comment:
What is the hasDuplicates state element doing? Are we allowing them to add cards more than once to their card list? 
I really like the use of find here!